### ISSUE TITLE
feature request(cli): lint directories

### ISSUE THREAD

**Issue by vegerot** [quick-lint/quick-lint-js#1167]
Problem:
I want to run quick-lint on a large project (120k files, 6m SLoC).

Approaches tried:
- `quick-lint-js $(fd ...)` zsh can't handle that many arguments
- manually calling `exec("/usr/local/bin/quick-lint-js", ...)` fails because it exceeds `NCARGS`
- writing my own script that batches calls to quick-lint in batch sizes of `NCARGS` works but is inconvenient to users

Recommendation:
quick-lint's CLI should also accept a directory.  quick-lint will recursively find all appropriate files (`.js`, `.ts`, etc.) and lint them.

```sh
$ quick-lint-js .
```

Out of scope:
Respect `.gitignore` files.  This is actually a requirement for my use-case because I don't want to lint `node_modules/`, `dist/`, and a bunch of hidden caches my build system creates.  This is also the default behavior of `fd`.

**Comment by vegerot** [comment]
note: #1166 is an alternative solution

**Comment by strager** [comment]
Duplicate of #221
